docs/design/asset-canister-interface.md

```diff
@@ -0,0 +1,508 @@
+# Asset Canister Interface
+
+## Introduction
+
+The asset canister interface provides for storage and retrieval of static assets, such as HTML, CSS, JavaScript, images, and other media files. It can store different content encodings of an asset's contents, such as `identity` and `gzip`.
+
+A canister that implements this interface can also return dynamic results from the [http_request](#method-http_request) method.
+
+This document is meant to describe the interface with enough detail to aid in understanding how the asset canister works and in interacting with the asset canister at the code level.  It does not describe the interface in sufficient detail to rise to the level of a specification.
+
+For brevity, this document does not reproduce the candid signatures for every method. 
+
+## Table of Contents
+
+| Section                                                  |
+|----------------------------------------------------------|
+| [Retrieving Assets](#retrieving-assets)                  |
+| [Storing Assets](#storing-assets)                        |
+| [Type Reference](#type-reference)                        |
+| [Method Reference](#method-reference)                    |
+| [Batch Operation Reference](#batch-operation-reference)  |
+| [Configuration Reference](#configuration-reference)      |
+| [Permission Reference](#permission-reference)            |
+
+## Retrieving Assets
+
+Asset retrieval begins with a call to either [get()](#method-get) or [http_request()](#method-http_request).
+
+### Asset Lookup
+
+The asset canister looks up assets by [key](#key) in a case-sensitive manner.
```

Comment:
It may be helpful to understand what a `key` is in this context. Is it a filename? Is it something generated by dfx? As a developer, why do I care about the key?

Reply:
Yeah, I struggled a bit with how much to define, and at what point, and how much to duplicate.  That's why `key` is a link here, to the field's definition in the "Type Reference."  I'll try adding some more prose here.

> Key
The key is a case-sensitive string that identifies the asset. By convention, all asset keys begin with a forward slash. For example, /index.html.